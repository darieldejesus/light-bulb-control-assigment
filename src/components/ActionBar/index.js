import React from 'react';
import { Button, Txt, Box } from 'rendition';
import styled from 'styled-components';
import FaHome from 'react-icons/lib/fa/home';
import FaChevronLeft from 'react-icons/lib/fa/chevron-left';
import FaLightBulbO from 'react-icons/lib/fa/lightbulb-o';

const FaLightBulb = styled(FaLightBulbO)`
  margin: -0.2em 0.4em 0 0;
`;

export default () => (
  <Box pt={2} pb={[3, 4]} pl={3}>
    <Button py={2} px={0} m={0} quartenary square color="tertiary.main">
      <FaHome />
    </Button>
    <Button py={[0, 1, 2]} px={3} ml={2} quartenary color="tertiary.main">
      <FaChevronLeft />
      <Txt.span ml={2} fontSize={[0, 1]}>
        Back
      </Txt.span>
    </Button>
    <Txt.span ml={4} fontSize={[2, 3]} color="tertiary.main">
      <FaLightBulb />
      Lighting
    </Txt.span>
  </Box>
);
