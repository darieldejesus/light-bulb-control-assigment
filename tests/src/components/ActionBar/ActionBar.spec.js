import React from 'react';

import ActionBar from '../../../../src/components/ActionBar';

describe('ActionBar', () => {
  test('Should render the component', () => {
    const wrapper = shallow(<ActionBar />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
