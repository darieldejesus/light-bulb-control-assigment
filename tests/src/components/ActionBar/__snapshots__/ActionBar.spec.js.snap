// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionBar Should render the component 1`] = `
<Component
  pb={
    Array [
      3,
      4,
    ]
  }
  pl={3}
  pt={2}
>
  <WithTheme(Component)
    color="tertiary.main"
    m={0}
    px={0}
    py={2}
    quartenary={true}
    square={true}
  >
    <FaHome />
  </WithTheme(Component)>
  <WithTheme(Component)
    color="tertiary.main"
    ml={2}
    px={3}
    py={
      Array [
        0,
        1,
        2,
      ]
    }
    quartenary={true}
  >
    <FaChevronLeft />
    <Component
      fontSize={
        Array [
          0,
          1,
        ]
      }
      ml={2}
    >
      Back
    </Component>
  </WithTheme(Component)>
  <Component
    color="tertiary.main"
    fontSize={
      Array [
        2,
        3,
      ]
    }
    ml={4}
  >
    <Styled(FaLightbulbO) />
    Lighting
  </Component>
</Component>
`;
